viewer/packages/3d-overlays/src/Overlay3D.ts

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright 2023 Cognite AS
+ */
+export interface Overlay3D<MetadataType> {
+  /**
+   * Set whether this overlay should be visible.
+   * */
+  set visible(visible: boolean);
+  /**
+   * Get whether this overlay is visible.
+   */
+  get visible(): boolean;
+  /**
+   * Get the position of this overlay.
+   */
+  get position(): THREE.Vector3;
+  /**
+   * Set the display color of this overlay.
+   * */
+  setColor(color: THREE.Color): void;
+  /**
+   * Get the display color of this overlay.
+   * */
+  get color(): THREE.Color;
+  /**
+   * Get the metadata associated with this overlay.
+   * */
+  getMetadata(): MetadataType | undefined;
```

Comment:
Should we allow the metadata to be undefined on a SmartOverlay? :thinking:

Reply:
Agree, I was initially thinking of making it undefined when there is nothing supplied in metadata from `OverlayInfo` but now I think it makes more sense to make it always defined and then the type decides what should be optional, i.e. is there a possibility for empty metadata.

Reply:
Okay, when I started implementing it, it turned out to be impossible without lying to typescript to have metadata optional when creating new overlay but always defined metadata field on actual overlay. So, we are either ok to require always passing metadata to create a new overlay or leave it as possibly undefined.

Reply:
I guess will leave it as undefined for now.